### PR TITLE
Add orbit_case::IsDirectory

### PR DESCRIPTION
**Pull request by antonrohr**
And replace unsafe variant in Process.cpp

Test: Unit Test

### CODE REVIEW

**Review by vickyliu-go4it — APPROVED** [review]
LGTM

File: src/OrbitBase/FileTest.cpp [hunk 1]
```diff
@@ -428,4 +428,15 @@ TEST(File, GetFileDateModified) {
...
 
+TEST(File, IsDirectory) {
+  auto tmp_file_or_error = TemporaryFile::Create();
+  ASSERT_THAT(tmp_file_or_error, HasNoError());
+
+  std::filesystem::path tmp_file_path = tmp_file_or_error.value().file_path();
+
+  EXPECT_THAT(IsDirectory(tmp_file_path), HasValue(false));
+
+  EXPECT_THAT(IsDirectory(tmp_file_path.parent_path()), HasValue(true));
```
Comment (dpallotti) [inline src/OrbitBase/FileTest.cpp:439]: I would add a case with a path to a non-existing file/directory.

File: src/OrbitBase/include/OrbitBase/File.h [hunk 2]
```diff
@@ -130,6 +130,7 @@ ErrorMessageOr<uint64_t> FileSize(const std::filesystem::path& file_path);
 ErrorMessageOr<std::vector<std::filesystem::path>> ListFilesInDirectory(
     const std::filesystem::path& directory);
 ErrorMessageOr<absl::Time> GetFileDateModified(const std::filesystem::path& path);
+ErrorMessageOr<bool> IsDirectory(const std::filesystem::path& path);
```
[the remark this answers is not in the document]
Comment (dpallotti) [reply src/OrbitBase/include/OrbitBase/File.h:133]: Note that `ErrorMessageOr` is already marked as `[[nodiscard]]`, so all the functions here are implicitly already `[[nodiscard]]` without the annotation.

**Review by dpallotti — APPROVED** [review]

File: src/OrbitBase/FileTest.cpp (hunk 1 above)
Comment (antonrohr) [reply src/OrbitBase/FileTest.cpp]: done

**Pull request merged**